pkg/controller/replication/conversion.go

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file contains adapters that convert between RC and RS,
+// as if ReplicationController were an older API version of ReplicaSet.
+// It allows ReplicaSetController to directly replace the old ReplicationManager,
+// which was previously a manually-maintained copy-paste of RSC.
+
+package replication
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	v1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	extensionsv1beta1client "k8s.io/client-go/kubernetes/typed/extensions/v1beta1"
+	v1listers "k8s.io/client-go/listers/core/v1"
+	extensionslisters "k8s.io/client-go/listers/extensions/v1beta1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	extensionsinternalv1beta1 "k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
```

Comment:
Do we need some unit test for conversion?

Reply:
The RC/RS conversion process itself is well tested now after #53164. The only thing left to test here would be, "Did we call the conversion function for each object?" For that, I think it's more meaningful to check that these shims work in context, using the RC integration tests I've just added.